core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -985,7 +985,7 @@ abstract class RDD[T: ClassTag](
   def subtract(
       other: RDD[T],
       p: Partitioner)(implicit ord: Ordering[T] = null): RDD[T] = withScope {
-    if (partitioner == Some(p)) {
+    if (partitioner.contains(p)) {
```

Comment:
Do we still support scala 2.10? If we do, this will fail compilation.

Reply:
Actually I think the previous code is more readable.